tm2/pkg/crypto/secp256k1/secp256k1_nocgo.go

```diff
@@ -34,37 +30,36 @@ func (pubKey PubKeySecp256k1) VerifyBytes(msg []byte, sigStr []byte) bool {
 	if len(sigStr) != 64 {
 		return false
 	}
-	pub, err := secp256k1.ParsePubKey(pubKey[:], secp256k1.S256())
+
+	pub, err := secp256k1.ParsePubKey(pubKey[:])
 	if err != nil {
 		return false
 	}
-	// parse the signature:
-	signature := signatureFromBytes(sigStr)
-	// Reject malleable signatures. libsecp256k1 does this check but btcec doesn't.
-	// see: https://github.com/ethereum/go-ethereum/blob/f9401ae011ddf7f8d2d95020b7446c17f8d98dc1/crypto/signature_nocgo.go#L90-L93
-	if signature.S.Cmp(secp256k1halfN) > 0 {
+
+	psig, ok := signatureFromBytes(sigStr)
+	if !ok {
 		return false
 	}
-	return signature.Verify(crypto.Sha256(msg), pub)
+
+	return psig.Verify(crypto.Sha256(msg), pub)
 }
 
 // Read Signature struct from R || S. Caller needs to ensure
 // that len(sigStr) == 64.
-func signatureFromBytes(sigStr []byte) *secp256k1.Signature {
-	return &secp256k1.Signature{
-		R: new(big.Int).SetBytes(sigStr[:32]),
-		S: new(big.Int).SetBytes(sigStr[32:64]),
+func signatureFromBytes(sigStr []byte) (*ecdsa.Signature, bool) {
+	// parse the signature:
+	var r, s secp256k1.ModNScalar
+	if r.SetByteSlice(sigStr[:32]) {
+		return nil, false // overflow
+	}
+	if s.SetByteSlice(sigStr[32:]) {
```

Comment:
```suggestion
	if s.SetByteSlice(sigStr[32:64]) {
```

I need to read more code, but previously it seemed limited to a specific usage, which I generally prefer.

Reply:
This code is limited to be used with the compact signature representation. Not forcing a 64-byte array is less error-prone IMHO.